From 982b9c659b407eb208ef307b02dd82c3b98c62f1 Mon Sep 17 00:00:00 2001
From: Mohamed Gaber <me@donn.website>
Date: Wed, 25 Jun 2025 11:54:30 +0300
Subject: [PATCH 1/2] cmake: remove stray usages of libabc within the codebase

These two CMakeLists.txt files in src/rmp use `libabc` unqualified, instead of ${ABC_LIBRARY} (in spite of the fact one of them already links ${ABC_LIBRARY})

Signed-off-by: Mohamed Gaber <me@donn.website>
---
 src/rmp/src/CMakeLists.txt      | 1 -
 src/rmp/test/cpp/CMakeLists.txt | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/rmp/src/CMakeLists.txt b/src/rmp/src/CMakeLists.txt
index 1c3b769cd06..9e1266c3b24 100644
--- a/src/rmp/src/CMakeLists.txt
+++ b/src/rmp/src/CMakeLists.txt
@@ -65,7 +65,6 @@ target_link_libraries(rmp_abc_library
   PUBLIC
     OpenSTA
     dbSta_lib
-    libabc
     utl_lib
     ${ABC_LIBRARY}
 )
diff --git a/src/rmp/test/cpp/CMakeLists.txt b/src/rmp/test/cpp/CMakeLists.txt
index 95dc2a3cdaf..35ec90b9163 100644
--- a/src/rmp/test/cpp/CMakeLists.txt
+++ b/src/rmp/test/cpp/CMakeLists.txt
@@ -9,7 +9,7 @@ target_link_libraries(RmpGTests
         rmp_abc_library
         dbSta_lib
         utl_lib
-        libabc
+        ${ABC_LIBRARY}
         ${TCL_LIBRARY}
 )
 

From f8fc878e81db06ba70590845863587bed3608e87 Mon Sep 17 00:00:00 2001
From: Mohamed Gaber <me@donn.website>
Date: Wed, 25 Jun 2025 13:03:38 +0300
Subject: [PATCH 2/2] cmake: make USE_SYSTEM_OPENSTA export a proper library

When USE_SYSTEM_OPENSTA is set, adding OpenSTA as a STATIC, IMPORTED library allows CUDD_LIB to be added to its INTERFACE_LINK_LIBRARIES property, meaning all of the test binaries linked against OpenSTA will automatically be linked against CUDD_LIB. This is not propagated automatically otherwise and CUDD_LIB would have to be added to every single one of them.

Signed-off-by: Mohamed Gaber <me@donn.website>
---
 src/CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c4d259f9207..6746cea6f53 100755
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -61,6 +61,15 @@ if(USE_SYSTEM_OPENSTA)
   endif()
   include_directories(${OPENSTA_INCLUDE_DIR})
   message(STATUS "Using OPENSTA_INCLUDE_DIR=${OPENSTA_INCLUDE_DIR}")
+  add_library(OpenSTA STATIC IMPORTED)
+  set_target_properties(OpenSTA PROPERTIES
+    IMPORTED_LOCATION "${OPENSTA_LIBRARY}"
+  )
+  if(CUDD_LIB)
+    set_target_properties(OpenSTA PROPERTIES 
+      INTERFACE_LINK_LIBRARIES "${CUDD_LIB}"
+    )
+  endif()
 else()
   # local build of opensta
   set(OPENSTA_LIBRARY "sta_swig")
